inference/core/cache/memory.py

```diff
@@ -149,3 +150,13 @@ def zremrangebyscore(
         for k in keys_to_delete:
             del self.cache[key][k]
         return len(keys_to_delete)
+
+    def acquire_lock(self, key: str) -> Any:
+        if key not in self.cache:
+            self.cache[key] = Lock()
+
+        lock = self.cache[key]
```

Comment:
Also, should we set an expiration on a lock? Something like 1 minute or something? I'm thinking of a case where a thread that has a lock dies not gracefully and then that model gets locked indefinitely.

Reply:
Set and defaulted to 3 minutes.